hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonFileInputFormat.java

```diff
@@ -208,6 +215,18 @@ public CarbonTable getOrCreateCarbonTable(Configuration configuration) throws IO
     return carbonFiles;
   }
 
+  private List<CarbonFile> getAllCarbonDataFiles(List fileLists) {
+    List<CarbonFile> carbonFiles = new LinkedList<CarbonFile>();
```

Comment:
Why you are using LinkedList??, array list will also be okay, please check

Reply:
for array, it's not convenient for adding new file  because it need change number of element.

Reply:
Better to change to arraylist insertion will be faster ....and memory footprint will be also lesser than linkedlist...., linked list will be good when u want to insert any element in between or you want to remove in between and both these operation are not done here

Reply:
has add operation for list:

```
  public CarbonReaderBuilder withFileLists(List fileLists) {
    if (null == this.fileLists) {
      this.fileLists = fileLists;
    } else {
      this.fileLists.addAll(fileLists);
    }
    return this;
  }
```

Reply:
Insert and add are different operation

Reply:
ok, changed.